From 22046fc0c9f6b26cf3bb72bb8d397823b569b31d Mon Sep 17 00:00:00 2001
From: Yael Samet <yael.samet@intel.com>
Date: Mon, 7 Jan 2019 09:01:24 +0200
Subject: [PATCH 0993/1225] mei: dal: dal_dev_read() - check for invalid
 argument

Need to dal_dev_read() if user the supplied buffer is not NULL.

Change-Id: I83ce529d9859090ddc959812a3f9c3abd5c376dd
Tracked-On: PKT-1711
Signed-off-by: Yael Samet <yael.samet@intel.com>
Signed-off-by: Tomas Winkler <tomas.winkler@intel.com>
---
 drivers/misc/mei/dal/dal_cdev.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/misc/mei/dal/dal_cdev.c b/drivers/misc/mei/dal/dal_cdev.c
index 48f209a..7b1aa55 100644
--- a/drivers/misc/mei/dal/dal_cdev.c
+++ b/drivers/misc/mei/dal/dal_cdev.c
@@ -107,6 +107,9 @@ static ssize_t dal_dev_read(struct file *fp, char __user *buf,
 	size_t r_len, len;
 	unsigned int copied;
 
+	if (!buf)
+		return -EINVAL;
+
 	ret = dal_wait_for_read(dc);
 	if (ret)
 		return ret;
-- 
2.7.4

